culinary-canvas/capstone-project/frontend/my_capstone_project/src/FindRecipes.jsx

```diff
@@ -0,0 +1,139 @@
+import { Link } from "react-router-dom";
+import React, { useState, useEffect } from "react";
+import { useAuth } from "./RedirectToAuthentication";
+import LoadingSpinner from "./LoadingSpinner";
+function FindRecipes() {
+  const { user } = useAuth();
+  const [ingredients, setIngredients] = useState(["", "", ""]);
+  const [recipes, setRecipes] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  const handleInputChange = (index, value) => {
+    const newIngredients = [...ingredients];
+    newIngredients[index] = value;
+    setIngredients(newIngredients);
+  };
+
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    setLoading(true);
+    try {
+      const response = await fetch("http://localhost:3000/recommendations", {
```

Comment:
Should we be hardcoding `localhost:3000` here? What if this app is later deployed on the web and the URL isn't `localhost:3000` but `culinary-canvas.com`? Can we use some type of relative link here?